### PR TITLE
Implement Fetching Of Comments

### PR DESCRIPTION
**Pull request by shainachen**
Added support for fetching comments with listings. Since each comment is tied to a listing, I added a comments list to the Listings class to hold all comments and added comment data to the HTTPResponse for fetching listing data.

### CODE REVIEW

File: src/main/java/com/google/roomies/data/Listing.java [hunk 1]
```diff
@@ -276,9 +294,29 @@ public Builder setListingPrice(String listingPrice) throws UnknownCurrencyExcept
...
+  private static ImmutableList<Comment> getAllCommentsFromCollection(String listingId)
+      throws IOException, InterruptedException, ExecutionException {
+    List<QueryDocumentSnapshot> documents = 
+      DatabaseFactory.getDatabase().getAllCommentsInListing(listingId).get().getDocuments();
+
+    List<Comment> comments =
+      StreamSupport.stream(documents.spliterator(), /* parallel= */ false)
+      .map(document -> Comment.fromFirestore(document))
+      .flatMap(Optional::stream)
+      .collect(Collectors.toList());
```
Comment (Snehalb26) [inline src/main/java/com/google/roomies/data/Listing.java:315]: You can directly use toImmutableList method from com.google.common.collect.ImmutableList.

File: src/main/java/com/google/roomies/data/Listing.java [hunk 2]
```diff
@@ -276,9 +294,29 @@ public Builder setListingPrice(String listingPrice) throws UnknownCurrencyExcept
...
+        .setComments(getAllCommentsFromCollection(document.getId()))
         .build());
     }
 
+  /**
+  * Gets all comments from a given listing.
+  * 
+  * @return list of comment instances
+  */
+  private static ImmutableList<Comment> getAllCommentsFromCollection(String listingId)
```
Comment (Snehalb26) [inline src/main/java/com/google/roomies/data/Listing.java:306]: This should be getAllCommentsForListing and may be rename getAllCommentsInListing which returns documents to 
getAllCommentDocumentsForListing.

Both the comments say same thing for these methods.

Comment (Snehalb26) [reply src/main/java/com/google/roomies/data/Listing.java:306]: Also, I don't see any test for Listing class.

File: src/main/java/com/google/roomies/data/Listing.java (hunk 1 above)
Comment (shainachen) [reply src/main/java/com/google/roomies/data/Listing.java:315]: Done

File: src/main/java/com/google/roomies/data/Listing.java (hunk 2 above)
Comment (shainachen) [reply src/main/java/com/google/roomies/data/Listing.java]: Renamed. Added tests for listing class and comments class.

File: src/test/java/com/google/roomies/data/CommentTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class CommentTest {
+  @Mock HttpServletRequest request;
+  @Mock QueryDocumentSnapshot queryDocumentSnapshotMock;
+
+  private static final String commentMessage = "Test message";
```
Comment (Snehalb26) [inline src/test/java/com/google/roomies/data/CommentTest.java:33]: Class constants should be all uppercase with words separated by underscores ("_").  ex. COMMENT_MESSAGE

File: src/test/java/com/google/roomies/data/ListingTest.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class ListingTest {
+  @Mock HttpServletRequest request;
+  @Mock QueryDocumentSnapshot queryDocumentSnapshotMock;
+
+  private static final String description = "Test description";
```
Comment (Snehalb26) [inline src/test/java/com/google/roomies/data/ListingTest.java:51]: Same as above naming convention for constants.

**Pull request merged**